aquarius/app/cached_block.py

```diff
@@ -0,0 +1,50 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+import logging
+import os
+from typing import Optional
+
+from flask_caching import Cache
+from aquarius.myapp import app
+
+logger = logging.getLogger(__name__)
+
+cache = Cache(
+    app,
+    config={
+        "CACHE_TYPE": "redis",
+        "CACHE_KEY_PREFIX": "ocean_aquarius",
+        "CACHE_REDIS_URL": os.getenv("REDIS_CONNECTION"),
+    },
+)
+
+
+def get_cached_block():
+    """
+    Retrieves the latest block value.
+    :return: `cached_block` stored in the database
+    """
+    return cache.get("cached_block")
+
+
+def update_cached_block(cached_block: Optional[int]):
+    """
+    Updates the value of `cached_block` in the database
+    :param: cached_block
+    """
+    if cached_block is None:
+        msg = "Cached block is None."
+        logger.error(msg)
+        raise Exception(msg)
+
+    block = get_or_create_cached_block(cached_block)
```

Comment:
you don't need this function. Please look at what you are copying and how you can adapt. The Nonce was "get_or_create" because we needed to initialize it when empty, depending on each user. Here, we can just set the cached block, simply.